diploma-project/components/ItemsScrollWithArrows.jsx

```diff
@@ -0,0 +1,96 @@
+import {useCallback, useEffect, useRef, useState} from "react";
+import {ChevronLeft, ChevronRight} from "react-feather";
+
+const ItemsScrollWithArrows = ({categoryName = "", listOfItems, message = ""}) => {
+    const [curr, setCurr] = useState(0);
+    const [translation, setTranslation] = useState(null);
+    const [timesToScroll, setTimesToScroll] = useState(null);
+    const containerRef = useRef(null);
+    const spacingInPixels = 20;
+
+    const prev = useCallback(() => setCurr((curr) => (curr === 0 ? 0 : curr - 1)), []);
+    const next = useCallback(() => setCurr((curr) => (curr === listOfItems.length - 1 ? listOfItems.length - 1  : curr + 1)), [listOfItems.length]);
+
+    const resetToFirst = () => setCurr(0);
+
+    useEffect(() => {
+        const updateCardWidth = () => {
+            if(listOfItems.length === 0) {
+                setTranslation(null);
+            } else if (containerRef.current) {
+                const cardWidth = containerRef.current.querySelector("#item-0").getBoundingClientRect().width;
+                if(cardWidth) {
+                    let itemScrollWidth = listOfItems.length * cardWidth + (listOfItems.length - 1) * spacingInPixels - containerRef.current.offsetWidth;
+                    if (itemScrollWidth > 0) {
+                        const temp = Math.ceil(itemScrollWidth / 680);
+                        setTimesToScroll(temp);
+                        setTranslation(itemScrollWidth / temp);
+                    } else {
+                        setTranslation(null);
+                    }
+                    resetToFirst(); // Reset to the first element on window resize
+                }
+            }
+        };
+
+        // Initial update
+        updateCardWidth();
+
+        // Listen for window resize and update card width
+        window.addEventListener("resize", updateCardWidth);
+
+        // Cleanup the event listener when the component unmounts
+        return () => {
+            window.removeEventListener("resize", updateCardWidth);
+        };
+    }, [listOfItems, containerRef.current]);
```

Comment:
<img width="606" alt="Screenshot 2024-02-12 at 19 25 40" src="https://github.com/ItsRizee/diploma_project/assets/42142769/f52dd480-eea0-485d-987f-0d2b5a869389">

Even the linter is complaining 🙃